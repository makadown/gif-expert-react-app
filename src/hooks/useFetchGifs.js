import { useEffect, useState } from "react";
import { getGifs } from "../helpers/getGifs";

/**
 * This hook takes a category as an argument.
 * Inside the hook, it uses the useState hook to create two state variables: images and isLoading.
 *  images is an array that will hold the fetched GIF images, and isLoading is a boolean that indicates whether the GIFs are still loading.
 * 
 * The getImages function is an asynchronous function that fetches GIFs using the getGifs function (presumably defined elsewhere) with the provided category. 
 * It then updates the images state with the fetched GIFs and sets isLoading to false.
 * 
 * The useEffect hook is used to call getImages when the component using this hook mounts. 
 * It does this by passing an empty dependency array [], which means the effect will only run once (on mount).
 * 
 * Finally, the hook returns an object with images and isLoading as its properties, 
 * which can be used by the component using this hook to render the GIFs and show a loading state if necessary.
 * @param {string} category - The category of gifs to fetch.
 * @return {Object} An object containing the fetched images and the loading state.
 */
export const useFetchGifs = (category) => {
    const [images, setImages] = useState([]);
    const [isLoading, setIsLoading] = useState(true);

    const getImages =  async() => {
        const newImages = await getGifs(category);
        setImages(newImages);
        setIsLoading(false);
    };

    useEffect(() => {
        getImages();
    }, []);

    return { images, 
            isLoading };
};