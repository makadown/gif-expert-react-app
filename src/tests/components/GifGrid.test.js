import React from 'react';
import { shallow } from 'enzyme';
import { useFetchGifs } from '../../hooks/useFetchGifs';
const { GifGrid } = require('../../components/GifGrid');

jest.mock('../../hooks/useFetchGifs');


describe('Pruebas de GifGrid', () => {
    const category = 'One Punch';

    test('Debe mostrar <GifGrid/> correctamente correctamente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid key={1} category={category} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {
        const gifs = [{
            id: "4N5vB4aErlVtVsywBw",
            url: "https://giphy.com/gifs/jojo-bizzare-cruisaders-4N5vB4aErlVtVsywBw",
            title: "bizzare GIF"
        }];
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
        expect( wrapper.find('p').exists()).toBe(false);
        expect( wrapper.find('GifGridItem').length ).toBe(gifs.length);
    });


});
