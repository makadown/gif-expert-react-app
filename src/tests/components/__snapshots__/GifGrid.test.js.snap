// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de GifGrid Debe de mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    One Punch
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="4N5vB4aErlVtVsywBw"
      key="4N5vB4aErlVtVsywBw"
      title="bizzare GIF"
      url="https://giphy.com/gifs/jojo-bizzare-cruisaders-4N5vB4aErlVtVsywBw"
    />
  </div>
</Fragment>
`;

exports[`Pruebas de GifGrid Debe mostrar <GifGrid/> correctamente correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    One Punch
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
